src/dmd/root/outbuffer.d

```diff
@@ -510,6 +539,80 @@ struct OutBuffer
             writeByte(0);
         return extractData();
     }
+
+    extern(D) bool moveToFile(const char* filename)
+    {
+        import dmd.root.file;
+        if (data == FileMapping!(const ubyte)(filename)[])
+        {
+            // Contents is the same, just touch the file
+            File.touch(filename);
```

Comment:
Ugh. All this drudgery related to OS files and mm files should be moved to FileMapping itself. The outbuffer shoud just say:

```d
if (fileMapping.active)
    fileMapping.moveToFile(file);
else
    write(data, file);
```

Will do this soonish.